radio/src/targets/common/avr/telemetry_driver.cpp

```diff
@@ -79,26 +94,177 @@ ISR(USART_RX_vect_N(TLM_USART))
-  RXD_PORT_N(TLM_USART) &= ~(1 << RXD_PORT_PIN_N(TLM_USART)); // disable pullup on RXD pin
+static void uart_4800(void) {
+  #undef BAUD
+  #define BAUD 4800
+  #include <util/setbaud.h>
+  UBRRH_N(TLM_USART) = UBRRH_VALUE;
+  UBRRL_N(TLM_USART) = UBRRL_VALUE;
+#if defined USE_2X
+  UCSRA_N(TLM_USART) |= (1 << U2X_N(TLM_USART)); // enable double speed operation.
+#else
+  UCSRA_N(TLM_USART) &= ~(1 << U2X_N(TLM_USART)); // disable double speed operation.
+#endif
+}
 
+static void uart_9600(void) {
   #undef BAUD
   #define BAUD 9600
   #include <util/setbaud.h>
+  UBRRH_N(TLM_USART) = UBRRH_VALUE;
+  UBRRL_N(TLM_USART) = UBRRL_VALUE;
+#if defined USE_2X
+  UCSRA_N(TLM_USART) |= (1 << U2X_N(TLM_USART)); // enable double speed operation.
+#else
+  UCSRA_N(TLM_USART) &= ~(1 << U2X_N(TLM_USART)); // disable double speed operation.
+#endif
+}
+
+static void uart_14400(void) {
+  #undef BAUD
+  #define BAUD 14400
+  #include <util/setbaud.h>
+  UBRRH_N(TLM_USART) = UBRRH_VALUE;
+  UBRRL_N(TLM_USART) = UBRRL_VALUE;
+#if defined USE_2X
+  UCSRA_N(TLM_USART) |= (1 << U2X_N(TLM_USART)); // enable double speed operation.
+#else
+  UCSRA_N(TLM_USART) &= ~(1 << U2X_N(TLM_USART)); // disable double speed operation.
+#endif
+}
 
+static void uart_19200(void) {
+  #undef BAUD
+  #define BAUD 19200
+  #include <util/setbaud.h>
   UBRRH_N(TLM_USART) = UBRRH_VALUE;
   UBRRL_N(TLM_USART) = UBRRL_VALUE;
+#if defined USE_2X
+  UCSRA_N(TLM_USART) |= (1 << U2X_N(TLM_USART)); // enable double speed operation.
+#else
   UCSRA_N(TLM_USART) &= ~(1 << U2X_N(TLM_USART)); // disable double speed operation.
+#endif
+}
+
+static void uart_38400(void) {
+  #undef BAUD
+  #define BAUD 38400
+  #include <util/setbaud.h>
+  UBRRH_N(TLM_USART) = UBRRH_VALUE;
+  UBRRL_N(TLM_USART) = UBRRL_VALUE;
+#if defined USE_2X
+  UCSRA_N(TLM_USART) |= (1 << U2X_N(TLM_USART)); // enable double speed operation.
+#else
+  UCSRA_N(TLM_USART) &= ~(1 << U2X_N(TLM_USART)); // disable double speed operation.
+#endif
+}
+
+static void uart_57600(void) {
+  #undef BAUD
+  #define BAUD 57600
+  #include <util/setbaud.h>
+  UBRRH_N(TLM_USART) = UBRRH_VALUE;
+  UBRRL_N(TLM_USART) = UBRRL_VALUE;
+#if defined USE_2X
+  UCSRA_N(TLM_USART) |= (1 << U2X_N(TLM_USART)); // enable double speed operation.
+#else
+  UCSRA_N(TLM_USART) &= ~(1 << U2X_N(TLM_USART)); // disable double speed operation.
+#endif
+}
+
+static void uart_58798(void) {
+  UBRRH_N(TLM_USART) = 0;
+  UBRRL_N(TLM_USART) = 0x010;
+  UCSRA_N(TLM_USART) &= ~(1 << U2X_N(TLM_USART)); // disable double speed operation.
+}
+
+static void uart_76800(void) {
+  #undef BAUD
+  #define BAUD 76800
+  #include <util/setbaud.h>
+  UBRRH_N(TLM_USART) = UBRRH_VALUE;
+  UBRRL_N(TLM_USART) = UBRRL_VALUE;
+#if defined USE_2X
+  UCSRA_N(TLM_USART) |= (1 << U2X_N(TLM_USART)); // enable double speed operation.
+#else
+  UCSRA_N(TLM_USART) &= ~(1 << U2X_N(TLM_USART)); // disable double speed operation.
+#endif
+}
+
+void telemetryPortInitFromIndex(uint8_t index) {
```

Comment:
Sorry I still don't follow are you saying you would prefer to kill the  "void telemetryPortInit(uint32_t baudrate)"  function? That is also required by "void telemetryPortInit()" which is used throughout. This telemetryPortInitFromIndex is just a workaround because the gui widget only returns the index to the option and not the value itself.
Sorry if I'm missing the obvious.